packages/ethers/src/client.ts

```diff
@@ -746,8 +746,18 @@ export class Web3Modal extends Web3ModalScaffold {
 
   private watchEmail() {
     if (this.emailProvider) {
-      this.emailProvider.onRpcRequest(() => {
-        super.open({ view: 'ApproveTransaction' })
+      this.emailProvider.onRpcRequest(request => {
+        const req = request as W3mFrameTypes.AppEvent & { payload?: unknown }
+        const payload = req.payload as W3mFrameTypes.RPCRequest
+        // We only open the modal if it's not a safe (auto-approve) or ignored method
+        const ALLOWED_METHODS = ['eth_blockNumber', 'eth_estimateGas', 'eth_getTransactionByHash']
```

Comment:
I think we should store these values in `W3mFrameConstants.ts`, so secure site and w3m are always in sync. Also better if we adjust these values in the future.